### PR TITLE
fix: Update error reponse doc for user update_user_profile and register

### LINKED ISSUE #927: Responses does not list username is invalid
**Describe the bug**

Here username is invalid is returned but not documented in the responses.

https://github.com/anitab-org/mentorship-backend/blob/0cf4d9de952edac8569adfaa9c08ae7defdf5a26/app/api/validations/user.py#L123-L126
ss

**To Reproduce**

1. Go to User
2. Scroll down to /user(updates user profile)
3. Add valid authorization token
4. Enter username as already used username
5. Click on execute
6. Scroll down to response

**Expected behavior**
The error messages should be included in the error responses

**Additional context**
Error mesaages for password, email and address should also be included

### PR DESCRIPTION
**Pull request by msklarek**
### Description

Updates to correct register and update user endpoints to show correct possible return error messages.

Fixes #927

### Type of Change:
- Code
- Documentation

**Code/Quality Assurance Only**
- Bugfix (non-breaking change which fixes an issue)

### How Has This Been Tested?

After code changed tested that the possible error response for the endpoints is shown correctly. 

image

image

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have made corresponding changes to the documentation

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
@msklarek thank you very much.
This PR looks good. Just use f-strings in your code.

**Comment by msklarek** [comment]
Fixed comments please review.  @devkapilbansal

**Review by devkapilbansal — COMMENTED** [review]
LGTM :+1: 
Will test it soon

Thanks @msklarek for this

**Comment by devkapilbansal** [comment]
@msklarek please enforce issue closing in the commit too

**Comment by msklarek** [comment]
@devkapilbansal I do not understand what you mean by "please enforce issue closing in the commit too"
Can you please explain a little more? Thanks.

**Comment by isabelcosta** [comment]
@msklarek I just changed markdown for mentioning the issue to close in your PR description, which changed from:
```
Fixes # https://github.com/anitab-org/mentorship-backend/issues/927
```
to:
```
Fixes #927
```

This will make GitHub close the issue mentioned when this PR is merged :) I think that was, what @devkapilbansal was trying to mention.

**Review by isabelcosta — DISMISSED** [review]
Great work @msklarek 👏 Thank you for your contribution and following our guidelines 🤗

**Comment by isabelcosta** [comment]
@devkapilbansal I changed the label to "Needs Testing", because you mentioned "LGTM". Please leave an approval if you approve the changes :)

**Comment by devkapilbansal** [comment]
> @devkapilbansal I changed the label to "Needs Testing", because you mentioned "LGTM". Please leave an approval if you approve the changes :)

I was testing this PR to ensure it doesn't break anything

**Review by devkapilbansal — DISMISSED** [review]
Tested :+1: 
The changes made in this PR were tested locally. Following are the results:

1. Code review - Done

2. All possible responses (positive and negative tests) were tested as below:

  * _Test1 Description_  List Error Responses in register endpoints
    _Screenshot/gif_:  

**Before Changes**
develop1
 
**After**
pr1

    _Expected Result_:  The endpoint should list all error responses
    _Actual Result_: Same as expected

  * _Test2 Description_  List Error Responses in Update User Endpoint
    _Screenshot/gif_:  

**Before Changes**
develop2

**After**
PR2

    _Expected Result_:  Endpoint should list all error responses in update endpoint
    _Actual Result_:  Same as expected
  
3. Additional testcases covered: Ran `python -m unittest discover tests`

4. Status of PR Changed to: Ready to Merge.

**Comment by devkapilbansal** [comment]
Looks like the last commit makes some merge conflicts. Please resolve these conflicts

**Comment by isabelcosta** [comment]
> > @devkapilbansal I changed the label to "Needs Testing", because you mentioned "LGTM". Please leave an approval if you approve the changes :)
> 
> I was testing this PR to ensure it doesn't break anything

Yes, I noticed. Thank you so much! your test is really detailed and amazing 👏 💯 🎉 @devkapilbansal

**Comment by gaurivn** [comment]
@msklarek, pls resolve the merge conflicts

**Comment by devkapilbansal** [comment]
@msklarek although the changes done here are good but you can see that the black checks are failing. Kindly run black formatter too and squash your commits in one

**Comment by msklarek** [comment]
I fixed the issues.

File: tests/tasks/test_api_create_task.py
Comment (devkapilbansal) [inline tests/tasks/test_api_create_task.py]: Write it link `f"/mentorship_relation/{self.mentorship_relation_w_second_user.id}/task`

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
All good just change all similar instances of strings I mentioned above

File: tests/tasks/test_api_create_task.py
Comment (isabelcosta) [inline tests/tasks/test_api_create_task.py]: Can you please remove the empty space in the string, like this and apply it to other examples here in the code?
```suggestion
            f"/mentorship_relation/{self.mentorship_relation_w_second_user.id}/task",
```

**Comment by isabelcosta** [comment]
@msklarek can you please address the change request, after that I would love to merge your work here, since this was already approved and tested 🎉

**Comment by msklarek** [comment]
@isabelcosta : I did the asked updated. The test on the branch seems to be failing but they look unrelated to my changes. Thanks.

**Comment by devkapilbansal** [comment]
@isabelcosta the errors are same as mentioned in #931 
Also, when I run the tests locally, they passed as they should. Can you please look into this?

**Comment by isabelcosta** [comment]
@msklarek can you update your branch with the latest changes from `develop` branch? If tests pass, I can approve 🚀 
@devkapilbansal may have solved the issue that was causing the tests to fail 🤞

**Comment by msklarek** [comment]
@isabelcosta  done!

**Review by isabelcosta — DISMISSED** [review]
thank you for fixing the merge conflicts!

**Comment by isabelcosta** [comment]
@vj-codes @devkapilbansal can you review this again? When I get a 2nd approval, I will merge this PR since it was already tested here https://github.com/anitab-org/mentorship-backend/pull/935#pullrequestreview-550863367

**Review by vj-codes — APPROVED** [review]
LGTM 🎉

**Review by vj-codes — CHANGES_REQUESTED** [review]
@msklarek hey there are som merge conflicts , can you please resolve then this PR is ready to merge 🚀

**Comment by devkapilbansal** [comment]
Also, @msklarek there are some responses that you missed like password accepting spaces. Please include them too

**Comment by msklarek** [comment]
Updated for the recently added code. Please have a look.

**Review by devkapilbansal — DISMISSED** [review]
LGTM :+1: 
Thanks @msklarek for this

@isabelcosta @isabelcosta can you please review it now

**Review by vj-codes — DISMISSED** [review]

File: app/api/resources/user.py
Comment (isabelcosta) [inline app/api/resources/user.py]: i think you have to use `.value` otherwise, it will show on Swagger UI as "HTTPStatus.BAD_REQUEST" instead of "400"

File: app/api/resources/user.py
Comment (isabelcosta) [inline app/api/resources/user.py]: same comment here:
> i think you have to use `.value` otherwise, it will show on Swagger UI as "HTTPStatus.BAD_REQUEST" instead of "400"

**Review by isabelcosta — CHANGES_REQUESTED** [review]
@msklarek can you please leave the `.value` on the lines you changed, if you take those out, Swagger will show in the response documentation HTTPStatus... instead of the actual number, i.e. 400.

After this change and merge conflicts are solved, I am happy to approve & get one more approval & test & and merge this :) 💪 thank you for this contribution!

**Comment by devkapilbansal** [comment]
@msklarek can you please look into this?

File: app/api/resources/user.py
Comment (msklarek) [reply app/api/resources/user.py]: Done

**Review by devkapilbansal — APPROVED** [review]
LGTM :+1: 
tagging @isabelcosta to review it again

**Comment by msklarek** [comment]
Fixed all comments.

**Review by isabelcosta — APPROVED** [review]
Great work @msklarek 🎉 
Thank you for contributing! 🙌

**Comment by isabelcosta** [comment]
I just double-checked on https://ms-backend-review-pr-935.herokuapp.com/
As a continuation of this test comment https://github.com/anitab-org/mentorship-backend/pull/935#pullrequestreview-550863367

Register API:

image

**Pull request merged**
